### ISSUE TITLE
Add documentation about security stance and hardening

### ISSUE THREAD

**Issue by josephlewis42** [earthly/earthly#942]
It would be awesome to have some docs describing the default security stance and threat model of the earthly tool as well as ways to harden it e.g. by using podman #760, or container runtimes like [gVisor](https://gvisor.dev/).

If I've missed this in the docs somewhere maybe the security docs could be un-nested?

**Comment by vladaionescu** [comment]
Makes sense - we don't have this yet. It would be great to show support for `runc` alternatives. #830 also requests this. Buildkit has configurable settings for this kind of stuff, but we've never tested if they work fine in Earthly.
